lib/bats-core/tracing.bash

```diff
@@ -119,15 +120,25 @@ bats_trim_filename() {
 
 bats_debug_trap() {
 	# don't update the trace within library functions or we get backtraces from inside traps
-	if [[ "$1" != $BATS_ROOT/lib/* && "$1" != $BATS_ROOT/libexec/* ]]; then
-		# The last entry in the stack trace is not useful when en error occured:
-		# It is either duplicated (kinda correct) or has wrong line number (Bash < 4.4)
-		# Therefore we capture the stacktrace but use it only after the next debug
-		# trap fired.
-		# Expansion is required for empty arrays which otherwise error
-		BATS_CURRENT_STACK_TRACE=("${BATS_STACK_TRACE[@]+"${BATS_STACK_TRACE[@]}"}")
-		bats_capture_stack_trace
-	fi
+	local stack_trace_offset=
+	if [[ "$1" == $BATS_ROOT/lib/* || "$1" == $BATS_ROOT/libexec/* ]]; then
+        	if [[ "${FUNCNAME[1]}" != "run" ]]; then
+			return
```

Comment:
The formatting seems to be a bit off here.

Reply:
It does not conform to the [formatting guidelines](https://github.com/bats-core/bats-core/blob/master/docs/CONTRIBUTING.md#formatting), but neither does the rest of that file. I chose to remain internally consistent within that context.

Reply:
Yes, the formatting is not consistent but in the preview the return is less indented than the if. Maybe this is a tabs vs spaces issue?